### ISSUE TITLE
Support for ZodPipeline

### ISSUE THREAD

**Issue by jesinmat** [asteasolutions/zod-to-openapi#115]
Hello,
I'm using Zod Pipelines according to the [Zod documentation](https://zod.dev/?id=you-can-use-pipe-to-fix-common-issues-with-zcoerce) to safely handle dates. I have the following object:

```
export const WorkerSchema = z
  .object({
    name: z.string().min(1),
    available: z.array(z.date().or(z.string().min(1).pipe(z.coerce.date()))),
  })
  .strict();
```

This way, I can either pass date directly or a string of length at least 1 that can be converted to a date.
Using this with zod-to-openapi, the following error is generated:

```
UnknownZodTypeError {
  message: 'Unknown zod object type, please specify `type` and other OpenAPI props using `ZodSchema.openapi`.',
  data: {
    currentSchema: { in: [ZodString], out: [ZodDate], typeName: 'ZodPipeline' },
    schemaName: undefined
  }
}
```

Since both ZodString and ZodDate are specified, is it possible to convert this automatically?
What should I do to overcome this issue in the meantime?
And since I use this approach several times in my project, can I specify this conversion method in one place for all of them?

**Comment by AGalabov** [comment]
Hi @jesinmat, thanks for using our library and pointing this out. `.pipe()` is relatively new so we've not looked into it before.

Overall I'd need to check if there is persistent way to make sure that the type is correctly inferred. For example `.transform` cannot be handled within `zod-to-openapi`. From what I see in the error I am optimistic.

As for a fix in the meantime - yes there is one. As the message of the error suggests you can just pass `.openapi({ type: 'string' ]})` to your schema. And for reusability I guess you can just use regular programming means and do something like:

```ts
// In some common file:
export const dateSchema = z.date().or(z.string().min(1).pipe(z.coerce.date()));

// In your current file:
export const WorkerSchema = z
  .object({
    name: z.string().min(1),
    available: z.array(dateSchema),
  })
  .strict();
```

Let me know if that works for you. And I'll see if we can provide an out of the box support for pipelines.

**Comment by jesinmat** [comment]
Thank you for your quick reply and the pull request!
I ended up using `.openapi({ type: "string", format: "date" })` and it works well. This library has helped me a lot, glad to see it's continuing to improve.

**Comment by AGalabov** [comment]
@jesinmat this should now be available in the [v4.6.0](https://github.com/asteasolutions/zod-to-openapi/releases/tag/v4.6.0) release
